### PR TITLE
[cleanup] remove unknown_params usage in preferences

### PR DESCRIPTION
**Pull request by glanham-jr**
## What does this PR do?

remove `unknown_params` usage in preferences - removing obsolete code. Also add some more unit tests to cover some basic cases for Preferences.

## Why is this change important?

Possible security issues in allowing arbitrary cookie manipulation originating not from SearXNG. Doesn't completely fix the issue, but a step towards mitigation.

More context from my comment here:

https://github.com/searxng/searxng/issues/3763#issuecomment-2424230198

## How to test this PR locally?

1. `make run`
2. Edit your browser cookies to add a random cookie not set by SearXNG
3. Verify that this does not show in Preferences

## Author's checklist

 - n/a

## Related issues

 - #3763

### CODE REVIEW

File: tests/unit/test_preferences.py [hunk 1]
```diff
@@ -155,8 +159,48 @@ def test_encode(self):
...
+    def test_false_key_value_setting(self):
+        setting_key = 'foo'
+
+        cookie_callback = {}
+
+        def set_cookie_callback(name, value, max_age):  # pylint: disable=unused-argument
+            cookie_callback[name] = value
+
+        response_mock = Mock(flask.Response)
+        response_mock.set_cookie = set_cookie_callback
```
Comment (Bnyro) [inline tests/unit/test_preferences.py:198]: I might be wrong, but wouldn't it be easier/cleaner to use a lambda function here?

Comment (glanham-jr) [reply tests/unit/test_preferences.py:198]: I was debugging with the lambda at one point and needed a multi line method - just didn't refactor it to a lambda. I'm indifferent if we want to in-line the function.

Comment (return42) [reply tests/unit/test_preferences.py:198]: In my opinion it is a callback function, and I prefer to read a clear callback definition (multi-line) over a more hidden inline (lambda) function.

@Bnyro, @glanham-jr: I haven't looked at the details of the PR yet, but is there anything else you think this PR shouldn't be merged yet?

Comment (Bnyro) [reply tests/unit/test_preferences.py:198]: Looks good to me.

**Review by Bnyro — APPROVED** [review]

**Pull request merged**